de4dot.code/deobfuscators/CryptoObfuscator/MethodsDecrypter.cs

```diff
@@ -54,7 +54,7 @@ class MethodsDecrypter {
 		bool Check(TypeDef type) {
 			if (type.NestedTypes.Count != 1)
 				return false;
-			if (type.Fields.Count != 3)
+			if (type.Fields.Count != 4)
```

Comment:
It might be better if you use this
if (type.Fields.Count < 3 || type.Fields.Count > 5)

Reply:
It will not work.

Reply:
Sample: 
[Sample.zip](https://github.com/mobile46/de4dot/files/9709739/Sample.zip)